### PR TITLE
Improve suite events handlers

### LINKED ISSUE #1528: suite.rc: add suite level events
#1503 adds in task runtime entries for [[[events]]] such as mail.

We should replicate this functionality for the suite level settings as well.

### PR DESCRIPTION
**Pull request by matthewrmshin**
Add built-in event email configurations.
Add general suite event handlers.
Allow flexible event handlers.
Run event handlers in process pool where possible.

Close #1528.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@arjclark @hjoliver please review.

**Comment by matthewrmshin** [comment]
(Branch re-based.)

File: lib/cylc/cfgspec/suite.py [hunk 1]
```diff
@@ -200,32 +200,24 @@ def _coerce_final_cycletime(value, keys, args):
...
-                default=GLOBAL_CFG.get([
-                    'cylc', 'event hooks', 'shutdown handler'])),
-            'timeout': vdr(
-                vtype='interval_minutes',
-                default=GLOBAL_CFG.get(['cylc', 'event hooks', 'timeout'])),
-            'reset timer': vdr(vtype='boolean', default=True),
+            'handlers': vdr(vtype='string_list'),
+            'handler events': vdr(vtype='string_list'),
+            'startup handler': vdr(vtype='string_list'),
+            'timeout handler': vdr(vtype='string_list'),
```
Comment (arjclark) [inline lib/cylc/cfgspec/suite.py]: No longer allowing setting a site default?

Comment (matthewrmshin) [inline lib/cylc/cfgspec/suite.py]: The logic has been moved to `Scheduler._get_events_conf` in `cylc.scheduler` for run time flexiblility.

Comment (arjclark) [inline lib/cylc/cfgspec/suite.py]: Cool. Ta.

**Comment by arjclark** [comment]
Looks ok to me. No problems from the test battery my end. Over to you @hjoliver

**Comment by hjoliver** [comment]
Branch needs updating.

**Comment by matthewrmshin** [comment]
Branch re-based.

**Comment by hjoliver** [comment]
Review 2: nice, no problems found, test battery passing here.

**Pull request merged**